test/unit/NodeCreateTransaction.js

```diff
@@ -0,0 +1,252 @@
+            AccountId.fromString("0.0.5006"),
+        ];
+        gossipCaCertificate = Buffer.from("gossipCaCertificate");
+        gossipEndpoints = [new ServiceEndpoint().setIpAddressV4(IP_AddressV4)];
+        serviceEndpoints = [new ServiceEndpoint().setIpAddressV4(IP_AddressV4)];
+
+        const TEST_ADMIN_KEY = PrivateKey.fromStringED25519(
+            "302e020100300506032b65700422042062c4b69e9f45a554e5424fb5a6fe5e6ac1f19ead31dc7718c2d980fd1f998d4b",
+        ).publicKey;
+
+        tx = new NodeCreateTransaction()
+            .setNodeAccountIds(nodeAccountIds)
+            .setTransactionId(
+                TransactionId.withValidStart(nodeAccountIds[0], VALID_START),
+            )
+            .setAccountId(accountId)
+            .setDescription(DESCRIPTION)
+            .setGossipCaCertificate(gossipCaCertificate)
+            .setCertificateHash(gossipCaCertificate) // TODO: change this vlaue to a proper one
+            .setAdminKey(TEST_ADMIN_KEY)
+            .setServiceEndpoints(serviceEndpoints)
+            .setGossipEndpoints(gossipEndpoints)
+            .setMaxTransactionFee(new Hbar(1));
+    });
+
+    it("should convert from and to bytes", function () {
+        const tx2 = NodeCreateTransaction.fromBytes(tx.toBytes());
+
+        tx.nodeAccountIds.forEach((element, index) => {
+            expect(tx.nodeAccountIds[index].toString()).to.equal(
+                tx2.nodeAccountIds[index].toString(),
+            );
+        });
+        expect(tx.transactionId.toString()).to.equal(
+            tx2.transactionId.toString(),
+        );
+        expect(tx.accountId.toString()).to.equal(tx2.accountId.toString());
+        expect(tx.description).to.equal(tx2.description);
+        expect(tx.gossipCaCertificate.equals(tx2.gossipCaCertificate)).to.be
+            .true;
+        expect(tx.certificateHash.equals(tx2.certificateHash)).to.be.true;
+        expect(tx.adminKey.toString()).to.equal(tx2.adminKey.toString());
+        expect(tx.maxTransactionFee.toTinybars().toInt()).to.equal(
+            tx2.maxTransactionFee.toTinybars().toInt(),
+        );
+    });
+
+    it("should change account id", function () {
+        const newAccountId = AccountId.fromString("0.4.20");
+        tx.setAccountId(newAccountId);
+        expect(tx.accountId.toString()).to.equal(newAccountId.toString());
+    });
+
+    it("should not change accound id if frozen", function () {
+        const newAccountId = AccountId.fromString("0.4.20");
+        tx.freeze();
+        let err = false;
+        try {
+            tx.setAccountId(newAccountId);
+        } catch (error) {
+            err = error.toString().includes("transaction is immutable");
+        }
+
+        if (!err) {
+            throw new Error(
+                "transaction successfully built from invalid bytes",
+            );
+        }
+    });
+
+    it("should change description", function () {
+        const NEW_DESCRIPTION = "New Description";
```

Comment:
If there's a `const` that has a hardcoded value there's a convention to use THIS_KIND_OF_NAMING. But for example there's also 
```javascript
const exampleObject = { a: 1, b:2}
exampleObject.a=3
```

Here people wouldn't use capital snake case naming because it isn't actually a constant. Yes, I know it says `const const exampleObject = { a: 1, b:2}` but you can still change it's value.

Reply:
Source: https://airbnb.io/javascript/#naming--uppercase